src/main/java/org/opentripplanner/routing/algorithm/mapping/RaptorPathToItineraryMapper.java

```diff
@@ -190,6 +190,21 @@ private Leg mapTransitLeg(
         leg.setIntermediateStops(
                 extractIntermediateStops(pathLeg, boardStopIndexInPattern, alightStopIndexInPattern));
 
+        if (tripSchedule.isFrequencyBasedTrip()) {
+            int headwaySeconds = tripSchedule.frequencyHeadwayInSeconds();
+            leg.setNonExactFrequency(headwaySeconds != 0);
+            leg.setHeadway(headwaySeconds);
+
+            // Fixup start time due to raptor handling of headways
+            Calendar startTime = leg.getStartTime();
+            startTime.setTimeInMillis(startTime.getTimeInMillis() + headwaySeconds * 1000L);
```

Comment:
Are these coming from Raptor with 'incorrect' start times, or is this a buffer added to ensure the connection can be made?

Reply:
I noticed a lot of buffering in the results when testing this code - which does make sense, so I'm not arguing against it, just curious.

Reply:
In the raptor timetable, the trip is boarded at the earliest possible time and alighted at the latest possible time, i.e. the driving time between the two stops + the headway, in case it is a non-exact frequency trip. This is to prevent a "too quick" boarding at the destination of the trip. In this mapper we then push the leg into the arrival end of that time span. 

In the new GraphQL API, we are going to split the Leg into multiple GraphQL types, and in the frequency based leg, I want to include two set of times, one earliest and one latest, so that the client can easily communicate the uncertainty of the times to the client.